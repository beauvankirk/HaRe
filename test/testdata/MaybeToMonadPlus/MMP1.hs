module MMP1 where

f :: Maybe Int -> Maybe Int
f Nothing = Nothing
f (Just i) = (Just (i + 1))
