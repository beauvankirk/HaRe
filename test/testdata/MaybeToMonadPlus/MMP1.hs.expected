module MMP1 where
import Control.Monad

f :: (MonadPlus m) => m Int -> m Int
f m_i = m_i >>= (\i -> (return (i + 1)))